wrapper/src/main/java/software/amazon/jdbc/plugin/failover/ReaderFailoverHandler.java

```diff
@@ -36,7 +37,8 @@ public interface ReaderFailoverHandler {
    * @return {@link ReaderFailoverResult} The results of this process.
    * @throws SQLException indicating whether the failover attempt was successful.
    */
-  ReaderFailoverResult failover(List<HostSpec> hosts, HostSpec currentHost) throws SQLException;
+  ReaderFailoverResult failover(
+      ConnectionService connectionService, List<HostSpec> hosts, HostSpec currentHost) throws SQLException;
```

Comment:
Can we pass ConnectionService as constructor parameter parameter to a class that implements this interface? I think the reference to ConnectionService never changes.